nri-prelude/src/Task.hs

```diff
@@ -3,6 +3,7 @@
 -- missing such a branch in a case statement looks like a problem and so is
 -- distracting.
 {-# OPTIONS_GHC -fno-warn-overlapping-patterns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
```

Comment:
it's not clear as to why we need this extension; we should be able to remove the type signatures in the let bindings within `background` and allow inference do its thing

Reply:
:curiosity: We actually do need it, cause otherwise won't be able to reconcile `background`s type variables with the ones we have inside the `let`. But it's already enabled as a default [here](https://github.com/NoRedInk/haskell-libraries/blob/1c996750f85816edbaee493cd92f6e22db94bb26/nri-prelude/package.yaml#L79)

Reply:
that's what i meant though - what happens if we remove the type declarations within the let?  should be ok, no?

Reply:
ooh, I see... probably, yeah
but I didn't want to leave it as inferred